openshift/fluentd.dc.yaml

```diff
@@ -0,0 +1,308 @@
+                failureThreshold: 3
+                # httpGet:
+                #   path: /
+                #   port: 8080
+                #   scheme: HTTP
+                exec:
+                  command:
+                  - "/bin/sh"
+                  - "-c"
+                  - "[[ $( pgrep ruby | wc -l)  == 2 ]]"
+                initialDelaySeconds: 10
+                periodSeconds: 60
+                successThreshold: 1
+                timeoutSeconds: 1
+              readinessProbe:
+                failureThreshold: 3
+                # httpGet:
+                #   path: /
+                #   port: 8080
+                #   scheme: HTTP
+                exec:
+                  command:
+                  - "/bin/sh"
+                  - "-c"
+                  - "[[ $( pgrep ruby | wc -l)  == 2 ]]"
+                initialDelaySeconds: 10
+                periodSeconds: 10
+                successThreshold: 1
+                timeoutSeconds: 1
+              name: "${APP_NAME}"
+              volumeMounts:
+                # mount fluentd config file
+                - name: fluentd-config-volume
+                  mountPath: /fluentd/etc/fluent.conf
+                  subPath: fluent.conf
+
+                # TLS cert for connecting to fluentd (enable when using TLS)
+                # - mountPath: /fluentd/etc/ssl
+                #   name: fluentd-tls
+              env:
+                - name: ${APP_NAME}-alert-endpoint
+                  valueFrom:
+                    secretKeyRef:
+                      name: ${APP_NAME}-alert
+                      key: alert-endpoint
+
+
+              ports:
+                - containerPort: 2020
+                  protocol: TCP
+                  name: metrics
+
+                - containerPort: 8080
+                  name: http-plugin
+                  protocol: TCP
+      test: false
+      triggers:
+        - type: ConfigChange
+parameters:
+  - name: APP_NAME
+    description: Application name
+    displayName: Application name
+    required: true
+    value: fluentd
+  - name: LABEL_ENV
+    description: Deployment environment
+    displayName: Environment
+    required: true
+    value: dev
+  - name: INSTANCE
+    description: The deployment instance name
+    displayName: Deployment Instance
+    required: true
+    value: master
+  - name: NAMESPACE
+    description: Target namespace reference (i.e. 'ixhmbm-dev')
+    displayName: Target Namespace
+    required: true
+    value: ixhmbm-dev
+  - name: HOST_ROUTE
+    description: The host the route will use to expose service outside cluster
+    displayName: Host route
+    required: true
+    value: fluentd-dev.pathfinder.gov.bc.ca
+  - name: CPU_LIMIT
+    description: Limit Peak CPU per pod (in millicores ex. 1000m)
+    displayName: CPU Limit
+    value: 1000m
+  - name: CPU_REQUEST
+    description: Requested CPU per pod (in millicores ex. 500m)
+    displayName: CPU Request
+    value: 100m
```

Comment:
Have you performed a cpu utilization analysis on fluentd? I'd double check to see if you really need 100m - otherwise try to drop it to 50m and see if it's still ok. I'd set the CPU limit to a lower number... perhaps 250m as I suspect this is not a compute-heavy process at all -> it's ok for this thing to run slow as we don't have any realtime pressure attached.